src/cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeMarshal.cs

```diff
@@ -0,0 +1,57 @@
+namespace System.Runtime.InteropServices.WindowsRuntime
+{
+    using System.Diagnostics;
+    using System.Diagnostics.CodeAnalysis;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::Windows.Foundation;
+    using global::Windows.Storage.Streams;
+    using WinRT;
+    /// <summary>
+    /// Contains extension methods that expose operations on WinRT <code>Windows.Foundation.IBuffer</code>.
+    /// </summary>
+#if EMBED
+    internal
+#else
+    public
+#endif
+    static partial class WindowsRuntimeMarshal
+    {
+        public static unsafe bool TryGetDataUnsafe(IBuffer buffer, out IntPtr dataPtr)

```

Comment:
Add some XML docs with also some remarks explaining why this is unsafe and how to use it safely?